### DISCUSSION TITLE
Is it useful to create a pool with 0 domains in it?

### DISCUSSION THREAD

**Discussion by kayceesrk** [ocaml-multicore/domainslib#85] [Q&A] [answered]
https://github.com/ocaml-multicore/domainslib/blob/master/lib/task.mli#L15

I can see that pool with 0 domains would work. But is that useful at all? @favonia @Sudha247.

**Comment by favonia** [comment, marked as answer]
I think it's useful when the pool size is a variable which could be zero. I can imagine cases where I don't want to additionally write a "sequential" version or where I want to measure the overhead introduced by domainslib.

**Reply by kayceesrk** [reply]
Agree with this sentiment partly. If you want to measure speedup, the correct baseline is a sequential version :-) But it may also be useful to see how much overhead domainslib running on 1 domain introduces over the sequential version (measuring pure overhead).

**Reply by favonia** [reply]
Yes, I meant the latter---comparing a traditional sequential version and a "parallel" version with a pool of size 0.
